src/cf.data.pre

```diff
@@ -5476,6 +5476,51 @@ DOC_START
 	rotated with "debug_options"
 DOC_END
 
+NAME: cache_log_message
+TYPE: cache_log_message
+DEFAULT: none
+DEFAULT_DOC: Use debug_options.
+LOC: TheDebugMessages
```

Comment:
Is it possible to make this array a part of the `Config` object? Isolated globals like this would be a bit harder to support during reconfiguration. I suspect that `Config` cannot be used in `Debug.h`, but the PR does not place the new code in `Debug.h` so perhaps we can use `Config`?

Reply:
In general I am against further extending SquidConf objects. The modularization of code into convenience libraries is about decreasing inter-dependencies. Adding to this massive widely used global goes against that.

This case is borderline since debug settings are likely to be widely needed anyway. IMO we should still keep it separate for its future home and life in libdebug and Debug:: namespace.

Reply:
> Amos: In general I am against further extending SquidConf objects. The modularization of code into convenience libraries is about decreasing inter-dependencies. Adding to this massive widely used global goes against that.

IMO, we should keep configuration in SquidConfig so that we can easily store/access two internally consistent and largely independent configurations during reconfigurations. Doing the same with a distributed collection of module-specific configuration objects would be a lot more difficult and uglier. That config-in-one-object ideal is far away from the current code, of course, but that is something we should move towards. Hence, my change request.

You are certainly right about inter-dependencies! That is one of the reasons we should structure SquidConfig so that it does _not_ introduce unnecessary inter-dependencies! For example, nobody using Config should know what Config.debugMessages really is unless they actually need access to that field (justifying the dependency). For everybody else, that field should be just a pointer to an object of a class with no known details.

Reply:
Thank you, I am resolving my change request.

@yadij, your "borderline" [comment](https://github.com/squid-cache/squid/pull/775#discussion_r576450012) did not feel like an objection to me, but if I misinterpreted your intent _and_ failed to convince you regarding the merits of keeping configuration in Config, please either unresolve this thread (or open your own change request) to discuss this further.